### DISCUSSION TITLE
Where to start with a Swift wrapper?

### DISCUSSION THREAD

**Discussion by justinfrench** [sfztools/sfizz#1063] [Q&A]
Hi folks, really love the SFZ format and excited to see there’s a C++ API for SFZ. Sampler options in the iOS ecosystem are pretty slim for drum libraries (there’s Apple’s AUSampler, with a poorly documented .aupreset file, and AudioKit’s Dunne sampler, which doesn’t have round robins, one-shots, etc).

So I’m what it would take to build a Swift wrapper around Sfizz, for use with AVAudioEngine or AudioKit.

Is anyone aware of existing projects or sample code I could reference?

**Comment by lucianoiam** [comment]
[ios-cmake](https://github.com/leetal/ios-cmake) can be used for building the needed static libraries.

Then an AudioUnit subclass written in ObjC/ObjC++ calls sfizz. Swift calling to the sfizz C API might work too but so far could not find conclusive information regarding the usage of Swift on the RT thread.

The AudioUnit can be then wrapped by a AVAudioUnit that fits AVAudioEngine or AudioKit.
